.github/workflows/workflow.yml

```diff
@@ -1,51 +1,99 @@
 name: Main workflow
 
 on:
+  workflow_dispatch:
   pull_request:
   push:
   schedule:
     # Prime the caches every Monday
     - cron: 0 1 * * MON
 
 jobs:
-  build:
+  build-test-non-domain-packages:
     strategy:
       fail-fast: false
       matrix:
         os:
-          - macos-latest
           - ubuntu-latest
+          - macos-latest
         ocaml-compiler:
-          - 4.08.x
-          - 4.09.x
-          - 4.10.x
-          - 4.11.x
-          - 4.12.x
-          - 4.13.x
+          - ocaml-variants.4.08.1+afl
+          - ocaml-variants.4.09.1+afl
+          - ocaml-variants.4.10.1+afl
+          - ocaml-variants.4.11.2+afl
+          - ocaml-variants.4.12.1+options,ocaml-option-afl
+          - ocaml-variants.4.13.1+options,ocaml-option-afl
+          - ocaml-variants.4.14.0+options,ocaml-option-afl
+        local-packages:
+          - |
+            *.opam
+            !cohttp-eio.opam
 
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Use OCaml ${{ matrix.ocaml-compiler }}
         uses: ocaml/setup-ocaml@v2
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
           dune-cache: ${{ matrix.os == 'ubuntu-latest' }}
-          opam-depext: true
+          opam-depext-flags: --with-test
+          opam-local-packages: ${{ matrix.local-packages }}
 
       - run: yarn --frozen-lockfile
         working-directory: cohttp_lwt_jsoo_test
 
       - run: echo "PKG_CONFIG_PATH=$(brew --prefix openssl)/lib/pkgconfig" >>"$GITHUB_ENV"
         if: ${{ matrix.os == 'macos-latest' }}
 
-      - run: opam install . --deps-only --with-test
+      - run: opam depext -i re uri uri-sexp sexplib0 ppx_sexp_conv stringext fmt alcotest conf-libev lwt cmdliner conduit-lwt-unix magic-mime logs ounit js_of_ocaml-lwt js_of_ocaml-ppx async async_unix base core core_unix conduit-async mirage-crypto ipaddr conduit-mirage mirage-kv astring base_quickcheck ppx_assert ppx_compare ppx_here core_bench crowbar ocurl mirage-channel lwt_ppx core_kernel.v0.14.2
```

Comment:
I'd like to avoid having to spell everything out by hand just for core-kernel or fix that. I can look at it in a separate PR if you want

Reply:
This should be fixed by https://github.com/mirage/ocaml-cohttp/pull/869
Once that is merged, you should be able to just use `- run: opam install . --deps-only --with-test`

Reply:
By the way, why using depext worked while using install broke? Opam 2.1 should install the depexts automatically, no?

Reply:
I also see why you want this, otherwise we are already installing plenty of cohttp packages already... argh

Reply:
Indeed, setup-ocaml defaults to using opam 2.0 where --with-deps --with-test doesn't seem to work when we have `http {=version}` in the opam files.